stdlib/public/CMakeLists.txt

```diff
@@ -19,6 +19,19 @@ if(SWIFT_STDLIB_SIL_DEBUGGING)
   list(APPEND SWIFT_RUNTIME_SWIFT_COMPILE_FLAGS "-Xfrontend" "-sil-based-debuginfo")
 endif()
 
+option(SWIFT_RUNTIME_PART_OF_DARWIN_SHARED_CACHE
+       "On Darwin platforms, link the standard library in a way that
+       allows inclusion in the dyld shared cache"
+       FALSE)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND
+   LLVM_LINKER_DETECTED AND
```

Comment:
Based on https://github.com/llvm/llvm-project/pull/77806/files#diff-f60dae8e82f67fa5e534e9ea566430d5b79f36119020865dc37b95d9a3635c5eR216, it seems like this should be based on `LLVM_LINKER_IS_APPLE` instead.

Reply:
Good observation -- in hindsight I wanted to be too clever, the flag is only properly understood by Apple linkers (LLD in fact ignores it).

Feedback taken in abac6822b2bf567e39f29c40b9e326024dd6c738 (and reworded commit description appropriately)